inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LpadFunction.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.util.List;
+
+/**
+ * LpadFunction
+ * description: Lpad(s1,len,s2) Fill string s2 at the beginning of string s1 to make the string length len
+ * - return null if any of the three parameters is null or len is less than 0
+ * - return "" if s2 is ""
+ * - return the substring of s1 with subscripts in the range of [0, len) if len is less than the length of s1
+ * - return the filled string
+ */
+public class LpadFunction implements ValueParser {
+
+    private final ValueParser leftStringParser;
+    private final ValueParser lengthParser;
+    private final ValueParser rightStringParser;
+
+    public LpadFunction(Function expr) {
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        leftStringParser = OperatorTools.buildParser(expressions.get(0));
+        lengthParser = OperatorTools.buildParser(expressions.get(1));
+        rightStringParser = OperatorTools.buildParser(expressions.get(2));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object leftStringObj = leftStringParser.parse(sourceData, rowIndex, context);
+        Object lengthObj = lengthParser.parse(sourceData, rowIndex, context);
+        Object rightStringObj = rightStringParser.parse(sourceData, rowIndex, context);
+        if (leftStringObj == null || lengthObj == null || rightStringObj == null) {
+            return null;
+        }
+        int len = Integer.parseInt(OperatorTools.parseString(lengthObj));
+        if (len < 0) {
+            return null;
+        }
+        String rightStr = OperatorTools.parseString(rightStringObj);
+        if (rightStr.isEmpty()) {
+            return "";
+        }
```

Comment:
Why this function return "" if s2 is "", is there any official documentation? For example, lpad("hi", 1, "") returns "h" may be more appropriate?

Reply:
Thank you for your inspection. The official MySQL documentation does not provide a detailed description of it. This is my negligence, as I did not consider the situation where len is less than s1 length during the testing process. Now modifications have been made to class comments and code logic, please check.